CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+# Contributing to Typst
+Thank you for considering to contribute to Typst. We want to foster a welcoming
+and productive atmosphere for contributers. Therefore, we outline a few steps to
+land your contribution below.
+
+1. Before starting significant work on a feature or refactoring, please
+   find/open an [issue] or start a thread in the [#contributors] channel on
+   Discord to discuss the design. Feel also free to ping a maintainer/team
+   member to get some input on your idea. Don't be shy! Typst is a complex
+   project with a long-term vision and it's frustrating to find out that your
+   idea does not align with that vision _after_ you have already implemented
+   something.
+2. Fork the Typst repository and start with your contribution. If you, at any
+   point in this process, are unsure about how to do something in the Typst
+   codebase, reach out to a maintainer or a more experienced contributor. Also
+   have a look at the [`ARCHITECTURE.md`][architecture] file. It explains how
```

Comment:
The definition of the link `architecture` in `CONTRIBUTING.md` is missing. Maybe add the following line (from `README.md`)?

```markdown
[architecture]: https://github.com/typst/typst/blob/main/ARCHITECTURE.md
```